src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -182,8 +183,9 @@ public function getRedirectResponse()
             throw new RuntimeException('This response does not support redirection.');
         }
 
+        /** @var $this RedirectResponseInterface */
         if ('GET' === $this->getRedirectMethod()) {
-            return HttpRedirectResponse::create($this->getRedirectUrl());
+            return new RedirectResponse($this->getRedirectUrl());
```

Comment:
We're not relying on the Diactoros here, we're specifying that we return a ResponseInterface. The Diactoros RedirectResponse is basically a factory that sets the correct location header + status on the object. See https://github.com/zendframework/zend-diactoros/blob/master/src/Response/RedirectResponse.php